pkg/rewards/plan.go

```diff
@@ -60,39 +66,81 @@ func (r *Plan) validate() error {
 			return errors.New("max participants must be positive in mechanics")
 		}
 		if len(mechanics.Tiers) > 1 {
-			for i := 1; i < len(mechanics.Tiers); i++ {
-				if mechanics.Tiers[i-1].MaxParticipants == mechanics.Tiers[i].MaxParticipants {
-					return fmt.Errorf("duplicate tier: %d in mechanics", mechanics.Tiers[i].MaxParticipants)
+			for j := 1; j < len(mechanics.Tiers); j++ {
+				if mechanics.Tiers[j-1].MaxParticipants == mechanics.Tiers[j].MaxParticipants {
+					return fmt.Errorf("duplicate tier: %d in mechanics", mechanics.Tiers[j].MaxParticipants)
 				}
 			}
 		}
+
+		// Check for conflicting redirects.
+		if len(mechanics.OwnerRedirects) > 0 && mechanics.OwnerRedirectsFile != "" {
+			return fmt.Errorf("both owner_redirects and owner_redirects_file specified for period %s", mechanics.Since)
+		}
+		if len(mechanics.ValidatorRedirects) > 0 && mechanics.ValidatorRedirectsFile != "" {
+			return fmt.Errorf("both validator_redirects and validator_redirects_file specified for period %s", mechanics.Since)
+		}
+
+		// Load CSV redirects if specified.
+		if mechanics.OwnerRedirectsFile != "" {
+			loadedRedirects, err := loadOwnerRedirectsFromCSV(mechanics.OwnerRedirectsFile)
+			if err != nil {
+				return fmt.Errorf("failed to load owner redirects from file %q: %w", mechanics.OwnerRedirectsFile, err)
+			}
+			mechanics.OwnerRedirects = loadedRedirects
+		}
+		if mechanics.ValidatorRedirectsFile != "" {
+			loadedRedirects, err := loadValidatorRedirectsFromCSV(mechanics.ValidatorRedirectsFile)
+			if err != nil {
+				return fmt.Errorf("failed to load validator redirects from file %q: %w", mechanics.ValidatorRedirectsFile, err)
+			}
+			mechanics.ValidatorRedirects = loadedRedirects
+		}
+
+		// Check for duplicate keys in OwnerRedirects.
+		ownerSeen := make(map[string]struct{}) // Separate map for OwnerRedirects
+		for from := range mechanics.OwnerRedirects {
+			if _, exists := ownerSeen[from.String()]; exists {
+				return fmt.Errorf("duplicate owner redirect key: %s", from.String())
+			}
+			ownerSeen[from.String()] = struct{}{}
+		}
+
+		// Check for duplicate keys in ValidatorRedirects.
+		validatorSeen := make(map[string]struct{}) // Separate map for ValidatorRedirects
+		for from := range mechanics.ValidatorRedirects {
+			if _, exists := validatorSeen[from.String()]; exists {
+				return fmt.Errorf("duplicate validator redirect key: %s", from.String())
+			}
+			validatorSeen[from.String()] = struct{}{}
+		}
 	}
```

Comment:
these checks will always pass as they're checking duplicate keys in a map (impossible)

i think the duplicate check can be in the load from CSV (where duplicates are possible)

Reply:
fixed